### ISSUE TITLE
Reference Resolver not resolving library-qualified identifiers correctly

### ISSUE THREAD

**Issue by ThunderFrame** [rubberduck-vba/Rubberduck#3301]
Omitting the `Global` qualifier leads to mixed results:
```vb
Sub test()
  'Host is Excel, Word is referenced
  Debug.Print Application.Name       'Application correctly resolves to Excel.Global.Application (property get accessor:Application)
  Debug.Print Excel.Application.Name 'Application INCORRECTLY resolves to Excel.IXmlSchema.Application (property get accessor:Object)
  Debug.Print Word.Application.Name  'Application INCORRECTLY resolves to Word.Email.Application (property get accessor:Application)
  Debug.Print Excel.Global.Application.Name 'Application correctly resolves to Excel.Global.Application (property get accessor:Application)
  Debug.Print Word.Global.Application.Name  'Application correctly resolves to Word.Global.Application (property get accessor:Application)
End Sub
```

In Outlook, the results are different again. The global property `Application` should resolve as a Property Get that returns an Application. Unlike Excel/Word, Outlook doesn't have a `Global` class, but members of the `Application` class are elevated as global members. In this case `Application.Application` is the global member.

![outlookapp](https://user-images.githubusercontent.com/16574009/29766199-4a7c645c-8c21-11e7-8e64-656c434daff3.png)

```vb
  Debug.Print Application.Name                'Application INCORRECTLY resolves to "Outlook.Application (Class)"
  Debug.Print Outlook.Application.Name        'Application INCORRECTLY resolves to "Outlook.Application (Class)"
  Debug.Print Application.Application.Name    'The 2nd Application almost correctly resolves to "Outlook.Application.Application (property get accessor:_Application)" - The return type should be `Application` not `_Application`?
```

**Comment by bclothier** [comment]
Just for the record, this requires us to do three things:

1) keep track of reference priority (this we already do somewhere else but I am not sure that the resolver is using it)
2) keep track of all objects that are flagged with [`appobject`](https://docs.microsoft.com/en-us/windows/win32/midl/appobject) attribute, which means it will have the `TYPEFLAG_FAPPOBJECT` flag for its type infos. That should be available via the `Com***` objects. As noted, not all agree on the naming conventions (e.g. Word and Excel uses `Global` but Outlook and Access uses `Application` as the `appobject`). 
3) resolve the unbound member access on the `appobject` as if it was a default member access but with the difference that any of members of the `appobject` can be implicitly accessed, and the reference priority impacts its accessibility -- meaning that if we have 2 `appobject`s loaded and they share same member, the one with higher priority effectively shadows the other's member, and can be only accessed by disambiguating with the containing library.

**Comment by MDoerner** [comment]
The reference resolver already uses the reference priority.

It seems as if the COM collector does not provide the necessary information yet.

**Comment by bclothier** [comment]
Indeed. We pick up the `control` attribute [here](https://github.com/rubberduck-vba/Rubberduck/blob/next/Rubberduck.Parsing/ComReflection/ComCoClass.cs#L63) but not for `appobject`. Because it's an attribute applied on a `coclass`, that means we need to then map it to the interfaces which is where we'll find the members. Judging from the documentation's silence, I'm assuming it means all the interfaces that coclass implements. 

For reference, [TYPEFLAG_FAPPOBJECT](https://docs.microsoft.com/en-us/windows/win32/api/oaidl/ne-oaidl-typeflags) has additional information in the Remarks section.
